### PR TITLE
OSX updates

### PR DESCRIPTION
**Pull request by slide**
Almost all tests should be passing now on OSX

### CODE REVIEW

File: Languages/IronPython/Tests/compat/common.py [hunk 1]
```diff
@@ -150,8 +150,8 @@ def get_platform_string(current = None):
     if current == None:
         import sys
         current = sys.platform
-        
-    if current.startswith("win") or current.startswith("linux"): return "nat"
+    
+    if [True for x in ["win", "linux", "darwin"] if current.startswith(x)]: return "nat"
```
Comment (slozier) [inline Languages/IronPython/Tests/compat/common.py]: Might be more readable if it was something like
`any(current.startswith(x) for x in ...)`

File: Languages/IronPython/Tests/modules/misc/_weakref_test.py [hunk 2]
```diff
@@ -43,6 +44,7 @@ def run_test():
         prxy = run_test()
         if not is_silverlight:
             gc.collect()
+            time.sleep(2)
```
Comment (slozier) [inline Languages/IronPython/Tests/modules/misc/_weakref_test.py]: Hmm, maybe only sleep in the Mono/osx case? Also a comment would be useful

File: Languages/IronPython/Tests/test_function.py [hunk 3]
```diff
@@ -787,11 +787,12 @@ def foo9(a,b): pass
...
     # check various function call sizes and boundaries
-    sizes = sizes = [3, 4, 5, 7, 8, 9, 13, 15, 16, 17, 23, 24, 25, 31, 32, 33, 47, 48, 49, 63, 64, 65, 127, 128, 129, 254, 255, 256, 257, 258, 511, 512, 513, 1023, 1024]
+    sizes = [3, 4, 5, 7, 8, 9, 13, 15, 16, 17, 23, 24, 25, 31, 32, 33, 47, 48, 49, 63, 64, 65, 127, 128, 129, 254, 255, 256, 257, 258, 511, 512, 513]
 
-    # mono has a limitation of 1024 for sizes
+
+    # mono has a limitation of < 1024
     if not is_mono:
-        sizes.extend([1025, 2047, 2048, 2049])
+        sizes.extend([1023, 1024, 1025, 2047, 2048, 2049])
```
Comment (slozier) [inline Languages/IronPython/Tests/test_function.py]: 1023 doesn't work? It's less than 1024. Maybe update the comment.

Comment (slide) [reply Languages/IronPython/Tests/test_function.py]: There is another item added, so 1023 + 1 = 1024, I can update the comment.

File: Languages/IronPython/Tests/compat/common.py (hunk 1 above)
Comment (slide) [reply Languages/IronPython/Tests/compat/common.py]: Yeah, that is better.

**Pull request merged**